arrow/src/test/java/org/apache/iceberg/arrow/vectorized/ArrowReaderTest.java

```diff
@@ -263,6 +265,60 @@ public void testReadColumnFilter2() throws Exception {
         scan, NUM_ROWS_PER_MONTH, 12 * NUM_ROWS_PER_MONTH, ImmutableList.of("timestamp"));
   }
 
+  @Test
+  public void testIssue10275() throws Exception {
```

Comment:
this should have a better method name. Something like `failsWithNPEWhenXyz`

Reply:
I've renamed the test to `testThrowsUOEWhenNewColumnHasNoValue`